test/CodeGenILTests/IndexingMicroTests.cs

```diff
@@ -0,0 +1,217 @@
+            Assert.AreEqual(5.0, output.Values.ToList()[2][0]);
+        }
+        [Test]
+        public void IndexIdentArrayWithIdent_MixedElementTypeArray()
+        {
+            var dscode = @"
+import(""DesignScriptBuiltin.dll"");
+a = [0.0,1.1,2.0,3.0,4.0,5.0,6,7,8,9,10];
+b = 1;
+c=a[b];";
+            var ast = ParserUtils.Parse(dscode).Body;
+            var output = codeGen.EmitAndExecute(ast);
+            Assert.IsNotEmpty(output);
+            Assert.AreEqual(1.1, output.Values.ToList()[2][0]);
+        }
+        [Test]
+        [Category("Failure")]//fails because we need to emit multiple ldelem opcodes per index.
+        public void IndexArrayWithArrayConstants()
+        {
+            var dscode = @"
+        import(""DesignScriptBuiltin.dll"");
+        a = [0,1,2,3,4,5];
+        b=a[[0,5]];";
+            var ast = ParserUtils.Parse(dscode).Body;
+            var output = codeGen.EmitAndExecute(ast);
+            Assert.IsNotEmpty(output);
+            CollectionAssert.AreEqual(new long[] { 0, 5 }, output.Values.ToList()[1][0] as long[]);
+        }
+        [Test]
+        public void IndexNestedArray_WithIdent()
+        {
+            var dscode = @"
+import(""DesignScriptBuiltin.dll"");
+a = [[0,1,2],[3,4,5]];
+b = 1;
+c = 0;
+d=a[b][c];";
+            var ast = ParserUtils.Parse(dscode).Body;
+            var output = codeGen.EmitAndExecute(ast);
+            Assert.IsNotEmpty(output);
+            Assert.AreEqual(3, output.Values.ToList()[3][0]);
+        }
+
+        [Test]
+        [Category("Failure")]//this fails because we can't index into an object
+                             //and [0..10] is assumed to be an object[] because it's generated with a function call.
+                             //this test actaully then fallsback correctly to ValueAtIndex but calls the wrong overload.
+        public void IndexIntoNestedRange_WithIdent()
+        {
+            var dscode = @"
+import(""DesignScriptBuiltin.dll"");
+a = [0..10]; //a is a []object
+b = 0;
+c = 5;
+d=a[b][c];";
+            var ast = ParserUtils.Parse(dscode).Body;
+            var output = codeGen.EmitAndExecute(ast);
+            Assert.IsNotEmpty(output);
+            Assert.AreEqual(4, output.Values.ToList()[3][0]);
+        }
+
+        [Test]
+        [Category("Failure")]//to accomplish this either need to generate a function call, or emit conditional il.
+        public void IndexIntoRange_WithNegativeIndex_Ident()
+        {
+            var dscode = @"
+import(""DesignScriptBuiltin.dll"");
+a = [0,1,2,3,4,5,6,7,8,9,10,11,12];
+b = -5;
+c=a[b];";
+            var ast = ParserUtils.Parse(dscode).Body;
+            var output = codeGen.EmitAndExecute(ast);
+            Assert.IsNotEmpty(output);
+            Assert.AreEqual(8, output.Values.ToList()[2][0]);
+        }
+        [Test]
+        [Category("Failure")]//I can't understand this failure, the IL is valid but the loading the element retrieves an IList instead of a string...
```

Comment:
Does your IL look like this:
![image](https://user-images.githubusercontent.com/5710686/181643246-94b4103d-4612-47be-b38c-2b28995918c2.png)

Reply:
yes, I actually think value is being loaded to the stack correctly, but stored in the output dictionary incorrectly somehow. I will try to write a test case that does not involve indexing but still fails.
https://jira.autodesk.com/browse/DYN-5156